src/transformers/modeling_tf_utils.py

```diff
@@ -31,7 +32,22 @@
 logger = logging.getLogger(__name__)
 
 
-class TFPreTrainedModel(tf.keras.Model):
+class TFModelUtils:
+    """
+    A few utilities for `tf.keras.Model`s, to be used as a mixin.
+    """
+
+    def num_parameters(self, only_trainable: bool = False) -> int:
+        """
+        Get number of (optionally, trainable) parameters in the model.
+        """
+        if only_trainable:
+            return int(sum(np.prod(w.shape.as_list()) for w in self.trainable_variables))
+        else:
+            return self.count_params()
```

Comment:
@jplu Does this look correct to you in TF/Keras?

Reply:
Sorry for the delay. I would do something more like the following:
```
import tensorflow.keras.backend as K

class TFModelUtils:

    def num_parameters(self) -> int:
        return self.num_non_trainable_parameters() + self.num_trainable_parameters()

    def num_non_trainable_parameters(self) -> int:
        return sum([K.count_params(w) for w in self.non_trainable_variables])

    def num_trainable_parameters(self) -> int:
        return sum([K.count_params(w) for w in self.trainable_variables])
```

Using the `count_params` function looks more elegant to me. Also, you can have simple functions to get the number of trainable and non trainable variables, but it is just my opinion your way is also ok :)